mlflow/tracing/fluent.py

```diff
@@ -750,22 +745,64 @@ def f():
     Returns:
         The last active trace if exists, otherwise None.
     """
-    trace_id = (
-        _LAST_ACTIVE_TRACE_ID_THREAD_LOCAL.get() if thread_local else _LAST_ACTIVE_TRACE_ID_GLOBAL
-    )
-    if trace_id is not None:
+    if _LAST_ACTIVE_TRACE_ID_GLOBAL is not None:
         try:
-            return MlflowClient().get_trace(trace_id, display=False)
+            return MlflowClient().get_trace(_LAST_ACTIVE_TRACE_ID_GLOBAL, display=False)
         except:
             _logger.debug(
-                f"Failed to get the last active trace with request ID {trace_id}.",
+                "Failed to get the last active trace with "
+                f"request ID {_LAST_ACTIVE_TRACE_ID_GLOBAL}.",
                 exc_info=True,
             )
             raise
     else:
         return None
 
 
+def get_last_active_trace_id(thread_local=False) -> Optional[Trace]:
```

Comment:
The return type hint here should be `Optional[str]` instead of `Optional[Trace]` since the function is returning the trace ID (a string) not the Trace object itself.